dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcContext.java

```diff
@@ -732,7 +732,11 @@ public <T> CompletableFuture<T> asyncCall(Callable<T> callable) {
                     if (o instanceof CompletableFuture) {
                         return (CompletableFuture<T>) o;
                     }
-                    return CompletableFuture.completedFuture(o);
+                    // not only local invoke will return directly, when return type of method is primitive type,
+                    // the proxy will convert null to default value of primitive type. therefore, should not return a completed Future when protocol is not injvm
+                    if (Constants.LOCAL_PROTOCOL.equals(url.getProtocol())) {
+                        return CompletableFuture.completedFuture(o);
+                    }
```

Comment:
If use async call invoke a local service without remote call, this will break the result